### PR TITLE
INTMDB-300: Advance Cluster resource ignoring the autoscaling options

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM, just needs internal reviewer.

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go [hunk 1]
```diff
@@ -842,6 +842,9 @@ func expandRegionConfigAutoScaling(tfList []interface{}) *matlas.AdvancedAutoSca
 		apiObject3.MaxInstanceSize = v.(string)
 	}
 
+	apiObject.DiskGB = apiObject2
+	apiObject.Compute = apiObject3
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_advanced_cluster.go]: This name is not helpful. Could we rename `apiObject` to `advancedAutoScaling`, `apiObject2` to `diskGB` and `apiObject3` to `compute` ?

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go [hunk 2]
```diff
@@ -842,6 +842,9 @@ func expandRegionConfigAutoScaling(tfList []interface{}) *matlas.AdvancedAutoSca
...
+
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_advanced_cluster.go:847]: Could we update `resource_mongodbatlas_advanced_cluster_test.go` to include the autoscaling attributes? Thanks!

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for all the changes

**Pull request merged**